lib/spack/spack/cmd/release_jobs.py

```diff
@@ -0,0 +1,138 @@
+##############################################################################
+# Copyright (c) 2013-2018, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+import os
+import sys
+import yaml
+
+import llnl.util.tty as tty
+from spack.error import SpackError
+from spack.util.spec_set import CombinatorialSpecSet
+
+description = "generate release build set as .gitlab-ci.yml"
+section = "build"
+level = "long"
+
+
+def setup_parser(subparser):
+    subparser.add_argument(
+        '-s', '--spec-set', default=None,
+        help="path to release spec-set yaml file")
+
+    subparser.add_argument(
+        '-m', '--mirror-url', default='http://172.17.0.1:8081/',
+        help="url of binary mirror where builds should be pushed")
+
+    subparser.add_argument(
+        '-o', '--output-file', default=".gitlab-ci.yml",
+        help="path to output file to write")
+
+
+def release_jobs(parser, args):
+    share_path = os.path.join('.', 'share', 'spack')
+    common_scripts_dir = os.path.join(share_path, 'docker', 'build', 'common')
+
+    os_container_mapping = {
+        'linux-ubuntu16.04-x86_64': {
+            'image': 'ubuntu:16.04',
+            'setup_script': os.path.join(common_scripts_dir,
+                'install-tools-ubuntu-16.04-%s.sh')
+        }
+    }
+
+    release_specs_path = args.spec_set
+    if not release_specs_path:
+        raise SpackError('Must provide path to release spec-set')
+
+    release_spec_set = None
+
+    with open(release_specs_path, 'r') as fin:
+        release_specs_contents = fin.read()
+        release_specs_yaml = yaml.load(release_specs_contents)
```

Comment:
minor point: consider using `spack_yaml` instead of `yaml`.  `spack_yaml` preserves file/line info from the file as attributes on the data structure it reads in (try typing `spack config blame config` if you are interested).